### DISCUSSION TITLE
hierholzer algorithm is not available

### DISCUSSION THREAD

**Discussion by Aswath98** [networkx/networkx#4785] [Q&A] [answered]
I need to implement hierholzer algorithm. Does networkx doesnt support it ? or is there any way I can implement this?

**Comment by dschult** [comment]
Any reference or link or hint as to what this person's name is associated with?   :}

**Reply by MatthewRalston** [reply]
Hello, also interested in [Hierholzers](https://en.m.wikipedia.org/wiki/Eulerian_path#Hierholzer's_algorithm). It's an algorithm for constructing Eulerian circuits in undirected graphs

**Reply by rossbar** [reply, marked as answer]
Finding Eulerian circuits is provided in the function `nx.eulerian_circuit`, defined [here](https://github.com/networkx/networkx/blob/5400927271b86c64f61b96e844462971a9d1d61b/networkx/algorithms/euler.py#L159).

The description of Hierholzer's algorithm on the Wiki page seems to align quite closely with the [implementation](https://github.com/networkx/networkx/blob/5400927271b86c64f61b96e844462971a9d1d61b/networkx/algorithms/euler.py#L112), at least at first glance. Whether the implementation is "Hierhozer-ian" enough for your use-case will depend on your specific criteria. However - if you want to find Eulerian circuits in Eulerian graphs, `nx.eulerian_circuit` is the place to start!

When searching the NetworkX docs, I'd recommend first searching by application (e.g. "Eulerian circuit") rather than by a specific named algorithm (e.g. "Hierhozers"). NetworkX *tends* to be organized by application/concept rather than by name!
